wavelength-plugins/digimon.js

```diff
@@ -0,0 +1,581 @@
+		};
+		let menu = `<div class="infobox"><center><strong>Digimon Search</strong></center><br/>`;
+		if (cmd !== 'digisearchdisplay') {
+			target = target.split(',').map(x => {
+				return x.trim();
+			});
+			for (let type in choices) {
+				menu += `<div><summary>${type.substring(0, 1).toUpperCase() + type.substring(1)}</summary>`;
+				for (let i = 0; i < choices[type].length; i++) {
+					let newTarget = false;
+					if (target.indexOf(type + ':' + choices[type][i]) > -1) {
+						newTarget = target.slice(0);
+						newTarget.splice(newTarget.indexOf(type + ':' + choices[type][i]), 1).join(',');
+					}
+					menu += `<button class="button" name="send" value="${newTarget ? `/searchchange ${newTarget}" style="background: #4286f4"` : `/searchchange ${target.join(',')}${change ? `,` : ``}${type}:${choices[type][i]}"`}>${choices[type][i]}</button>`;
+				}
+				menu += `</div>`;
+			}
+			if (toId(target.join(''))) {
+				// Show found digimon
+				let reqs = {alphabetical: '', stage: '', type: '', color: '', ability: ''};
+				let invalidSearch = false;
+				target.map(y => {
+					if (invalidSearch) return y;
+					y = y.split(':');
+					if (choices[y[0]] && choices[y[0]].includes(y[1])) {
+						if (reqs[y[0]]) {
+							invalidSearch = true;
+							return y.join(':');
+						}
+						reqs[y[0]] = y[1];
+					}
+					return y.join(':');
+				});
+				if (invalidSearch) {
+					menu += `No digimon were found. (Your search was invalid)</div>`;
+					return user.sendTo(room, `${change ? `|uhtmlchange|cs${user.userid}|` : `|uhtml|cs${user.userid}|`}${menu}`);
+				}
+				menu += `<center><div style='max-height: 300px; overflow-y: scroll;'>`;
+				let foundDigimon = 0;
+				for (const digimon in mod.data.Pokedex) {
+					let template = mod.getTemplate(digimon);
+					if (template.num > -2000 || template.num < -3000) continue;
+					if (reqs.alphabetical && !template.id.startsWith(reqs.alphabetical)) continue;
+					if (reqs.stage && reqs.stage !== template.stage) continue;
+					if (reqs.type && !template.types.includes(reqs.type)) continue;
+					if (reqs.color && reqs.color !== template.color) continue;
+					if (reqs.ability && reqs.ability !== template.abilities[0]) continue;
+					// Valid
+					foundDigimon++;
+					let first = 0;
+					let second = 0;
+					switch (template.id) {
+					case 'meicrackmonviciousmode':
+						first = 'Meicrackmon';
+						second = '[Vicious Mode]';
+						break;
+					case 'cherubimonevil':
+						first = 'Cherubimon';
+						second = '[Evil]';
+						break;
+					case 'cherubimongood':
+						first = 'Cherubimon';
+						second = '[Good]';
+						break;
+					case 'metalgreymonvaccine':
+						first = 'MetalGreymon';
+						second = '[Vaccine]';
+						break;
+					case 'metalgreymonvirus':
+						first = 'MetalGreymon';
+						second = '[Virus]';
+						break;
+					}
+					let type1 = 0;
+					switch (template.types[0]) {
+					case 'Flame':
+						type1 = 'https://i.imgur.com/a6oFkDS.png';
```

Comment:
imgur blocked showdown (when trying to load images on showdown web pages, the image will not load with a http 403 error), use the images that zarel uploaded for digimon instead (iirc there are some for the types as well).

Reply:
~~or if needed use discord to host ;X~~